ros2action/ros2action/api/__init__.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ament_index_python import get_resource
+from ament_index_python import get_resources
+from ament_index_python import has_resource
+
+import rclpy.action
+from rclpy.expand_topic_name import expand_topic_name
+from rclpy.validate_full_topic_name import validate_full_topic_name
+from ros2cli.node.direct import DirectNode
+
+
+def _is_action_status_topic(topic_name, action_name):
+    return action_name + '/_action/status' == topic_name
+
+
+def get_action_clients_and_servers(*, node, action_name):
+    action_clients = []
+    action_servers = []
+
+    try:
+        expanded_name = expand_topic_name(action_name, node.get_name(), node.get_namespace())
+    except ValueError as e:
+        raise RuntimeError(e)
+    try:
+        validate_full_topic_name(expanded_name)
+    except rclpy.exceptions.InvalidTopicNameException as e:
+        raise RuntimeError(e)
```

Comment:
Why raise a generic `RuntimeError` in this API? For the actual CLI I see why it should be a `RuntimeError` (to suppress the stacktrace) but for the API itself it would be more helpful to raise the specific exceptions.

So I would suggest to move the `try` / `except` logic to the verb using the API.

Reply:
Makes sense. Moved bbd8893